File: Frontend/lib/models/register_request.dart
class RegisterRequest {
  final String name;
  final String email;
  final String phone;
  final String password;
  final String role;

  RegisterRequest({
    required this.name,
    required this.email,
    required this.phone,
    required this.password,
    required this.role,
  });

  Map<String, dynamic> toJson() {
    return {
      "name": name,
      "email": email,
      "phone": phone,
      "password": password,
      "role": role,
    };
  }

  factory RegisterRequest.fromJson(Map<String, dynamic> json) {
    return RegisterRequest(
      name: json["name"] ?? "",
      email: json["email"] ?? "",
      phone: json["phone"] ?? "",
      password: json["password"] ?? "",
      role: json["role"] ?? "",
    );
  }
}

File: Backend/roomnest/src/main/java/com/roomnest/dto/RegisterRequest.java
package com.roomnest.dto;

import com.roomnest.enums.Role;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RegisterRequest {

    private String name;
    private String email;
    private String phone;
    private String password;
    private Role role;

}